src/Social/Contract/Comments/CommentsTrait.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Kanvas\Packages\Social\Contract\Comments;
+
+use Phalcon\Http\Response;
+use Kanvas\Packages\Social\Services\Comments;
+use Gewaer\Models\MessageComments;
+use Baka\Validation as CanvasValidation;
+use Phalcon\Validation\Validator\PresenceOf;
+
+/**
+ * Channels Trait
+ */
+trait CommentsTrait
+{
+    /**
+     * Get all the comments
+     *
+     * @param int $messageId
+     * @throws Exception
+     *
+     * @return Response
+     */
+    public function getAllComments(int $messageId) : Response
+    {
+        return $this->response($this->processOutput(MessageComments::findOrFail([
+            "conditions" => "message_id = :message_id: and apps_id = :apps_id: and companies_id = :companies_id: and is_deleted = 0",
+            "bind" => [
+                "message_id" => $messageId,
+                "apps_id" => $this->app->getId(),
+                "companies_id" =>$this->userData->getCurrentCompany()->getId(),
+            ]
+        ])));
+    }
+
+    /**
+     * Get comment by its id
+     *
+     * @param int $commentId
+     * @return Response
+     */
+    public function getComment(int $commentId): Response
+    {
+        return $this->response(Comments::getById((string)$commentId));
+    }
+
+    /**
+     * processInput function.
+     *
+     * @param array $request
+     * @return array
+     */
+    protected function processInput(array $request): array
+    {
+        $validation = new CanvasValidation();
+        $validation->add('message', new PresenceOf(['message' => _('message is required.')]));
+        $validation->validate($request);
+
+        return $request;
+    }
+
+    /**
+     * Add a new comment to a message
+     *
+     * @param int $messageId
+     * @return Response
+     */
+    public function addComment(int $messageId): Response
+    {
+        $request = $this->processInput($this->request->getPostData());
+        $newComment = Comments::add($messageId, $request['message']);
```

Comment:
@rwhite27 on before input we need to force when creating the user id, company id and app id

Reply:
we already do, those are set in the comments function which is called inside the add function: 

`/**
     * Create a comment for a message
     *
     * @param string $messageId
     * @param string $message
     * @return MessageComments
     */
    public function comment(string $message): MessageComments
    {
        $comment = new MessageComments();
        $comment->message_id = $this->getId();
        $comment->apps_id = Di::getDefault()->get('app')->getId();
        $comment->companies_id = Di::getDefault()->get('userData')->getDefaultCompany()->getId();
        $comment->users_id = Di::getDefault()->get('userData')->getId();
        $comment->message = $message;
        $comment->saveOrFail();

        return $comment;
    }`

Reply:
I would change this to provide the user as a param, that way I can use that function in other places not only relying on the DI

Reply:
@rwhite27 we still need to pass wthe user right?